pkg/upgraders/aroupgrader.go

```diff
@@ -0,0 +1,88 @@
+package upgraders
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	upgradev1alpha1 "github.com/openshift/managed-upgrade-operator/pkg/apis/upgrade/v1alpha1"
+	ac "github.com/openshift/managed-upgrade-operator/pkg/availabilitychecks"
+	cv "github.com/openshift/managed-upgrade-operator/pkg/clusterversion"
+	"github.com/openshift/managed-upgrade-operator/pkg/configmanager"
+	"github.com/openshift/managed-upgrade-operator/pkg/drain"
+	"github.com/openshift/managed-upgrade-operator/pkg/eventmanager"
+	"github.com/openshift/managed-upgrade-operator/pkg/machinery"
+	"github.com/openshift/managed-upgrade-operator/pkg/maintenance"
+	"github.com/openshift/managed-upgrade-operator/pkg/metrics"
+	"github.com/openshift/managed-upgrade-operator/pkg/scaler"
+	"github.com/openshift/managed-upgrade-operator/pkg/upgradesteps"
+)
+
+// aroUpgrader is a cluster upgrader suitable for ARO clusters.
+// It inherits from the base clusterUpgrader.
+type aroUpgrader struct {
+	*clusterUpgrader
+}
+
+// NewAROUpgrader creates a new instance of an aroUpgrader
+func NewAROUpgrader(c client.Client, cfm configmanager.ConfigManager, mc metrics.Metrics, notifier eventmanager.EventManager) (*aroUpgrader, error) {
+	cfg := &upgraderConfig{}
+	err := cfm.Into(cfg)
+	if err != nil {
+		return nil, err
+	}
+
+	m, err := maintenance.NewBuilder().NewClient(c)
+	if err != nil {
```

Comment:
just curious, what could be the err here?

Reply:
If the cluster has no alertmanager route or SA secret token, it would likely generate an error at this point in building up the AM client.